ovos_utils/location.py

```
@@ -0,0 +1,202 @@
+    try:
+        if not (-90 <= float(lat) <= 90) or not (-180 <= float(lon) <= 180):
+            raise ValueError("Invalid coordinates")
+        tz = TimezoneFinder().timezone_at(lng=float(lon), lat=float(lat))
+        if not tz:
+            raise RuntimeError(f"Failed to determine timezone from lat/lon: {lat}, {lon}")
+        return {
+            "name": tz.replace("/", " "),
+            "code": tz
+        }
+    except Exception as e:
+        raise ValueError(f"Invalid coordinates: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)